mne/epochs.py

```diff
@@ -87,9 +88,20 @@ def _save_split(epochs, fname, part_idx, n_parts):
     write_int(fid, FIFF.FIFF_MNE_EVENT_LIST, epochs.events.T)
     mapping_ = ';'.join([k + ':' + str(v) for k, v in
                          epochs.event_id.items()])
+
     write_string(fid, FIFF.FIFF_DESCRIPTION, mapping_)
     end_block(fid, FIFF.FIFFB_MNE_EVENTS)
 
+    # Metadata
+    if epochs.metadata is not None:
+        start_block(fid, FIFF.FIFFB_MNE_METADATA)
+        pd = _check_pandas_installed()
+        if not isinstance(epochs.metadata, pd.DataFrame):
+            raise ValueError('epochs.metadata must be a DataFrame')
```

Comment:
we shouldn't need to check this here. Use properties `@property` and setters `@metadata.setter` to refer to a `epochs._metadata`, and check for proper DataFrame-ness in the setter.